tortoise/util.go

```diff
@@ -135,3 +137,61 @@ func maxLayer(i, j types.LayerID) types.LayerID {
 	}
 	return j
 }
+
+func verifyLayer(logger log.Log, blocks []blockInfo, validity map[types.BlockID]sign, getDecision func(blockInfo) sign) bool {
+	// order blocks by height in ascending order
+	// if there is a support before any abstain
+	// and a previous height is lower than the current one
+	// the layer is verified
+	//
+	// it will modify original slice
+	sort.Slice(blocks, func(i, j int) bool {
+		return blocks[i].height < blocks[j].height
+	})
+	var (
+		decisions = make([]sign, 0, len(blocks))
+		supported blockInfo
+		positive  bool
+	)
+	for _, block := range blocks {
+		decision := getDecision(block)
+		logger.With().Debug("decision for a block",
+			log.Stringer("decision", decision),
+			log.Stringer("id", block.id),
+			log.Stringer("weight", block.weight),
+			log.Uint64("height", block.height),
+		)
+		if decision == abstain {
+			// all blocks with the same height should be finalized
+			if positive && block.height > supported.height {
+				decision = against
+			} else {
+				return false
+			}
+		} else if decision == support {
+			positive = true
+			supported = block
```

Comment:
this will overwrite the previous supported in the case where we support multiple blocks in the same layer.
are you saying that if we support multiple blocks in the same layer, they must have the same height?

Reply:
the point here is to wait until all blocks with the same height are verified, because the same weight should vote for them